src/wp-includes/block-template-utils.php

```diff
@@ -786,13 +777,55 @@ function _build_block_template_result_from_post( $post ) {
 		$template->is_custom = false;
 	}
 
+	return $template;
+}
+
+/**
+ * Builds a unified template object based a post Object.
+ *
+ * @since 5.9.0
+ * @since 6.3.0 Added `modified` property to template objects.
+ * @since 6.4.0 Added support for a revision post to be passed to this function.
+ * @access private
+ *
+ * @param WP_Post $post Template post.
+ * @return WP_Block_Template|WP_Error Template or error object.
+ */
+function _build_block_template_result_from_post( $post ) {
+	$post_id = wp_is_post_revision( $post );
+	if ( ! $post_id ) {
+		$post_id = $post;
+	}
+	$parent_post = get_post( $post_id );
+
+	$terms = get_the_terms( $parent_post, 'wp_theme' );
+
+	if ( is_wp_error( $terms ) ) {
+		return $terms;
+	}
+
+	if ( ! $terms ) {
+		return new WP_Error( 'template_missing_theme', __( 'No theme is defined for this template.' ) );
+	}
+
+	$terms = array(
+		'wp_theme' => $terms[0]->name,
+	);
+
 	if ( 'wp_template_part' === $parent_post->post_type ) {
 		$type_terms = get_the_terms( $parent_post, 'wp_template_part_area' );
 		if ( ! is_wp_error( $type_terms ) && false !== $type_terms ) {
-			$template->area = $type_terms[0]->name;
+			$terms['area'] = $type_terms[0]->name;
 		}
 	}
 
+	$meta = array(
+		'origin'           => get_post_meta( $parent_post->ID, 'origin', true ),
+		'is_wp_suggestion' => get_post_meta( $parent_post->ID, 'is_wp_suggestion', true ),
+	);
+
+	$template = _build_block_template_object_from_wp_post_object( $post, $terms, $meta );
```

Comment:
Since `_build_block_template_object_from_wp_post_object` can return a `WP_Error` shall we handle that here before attempting to assign further values to `$template`?

Reply:
Yeah, probably. The usual `if ( is_wp_error( $template ) ) { return $template; }` I guess?

Reply:
Yeah I think so. Given that further up in that function we're essentially doing the same thing by [returning a `WP_Error`](https://github.com/WordPress/wordpress-develop/pull/6278/files/5af11dba5ed4cf7c98fac4acfa8397e62e456b0a#diff-1d8e05d3ea69557ca8742330e62c1893a72d4ca4ff2ed4ccc5b94bcf5d20fed7R803-R809) that seems consistent with expected behaviour.

Reply:
https://github.com/WordPress/wordpress-develop/pull/6278/commits/a2fd228bd07619618f2edadf5a84b90785a1e650